### ISSUE TITLE
.htaccess in  URL cause 404 error

### ISSUE THREAD

**Issue by scorp13** [serghey-rodin/vesta#1889]
### VestaCP Version:
 0.9.8-24

In the default nginx configuration there is a line
location ~ /\.ht {return 404;} (if I understand correctly, in order to limit access to files starting with .ht)
however, I ran into a problem that when this combination is present in an URL, this rule also triggered.
So, when we try to open the example.com/.htaccess/some-htaccess-related-topic/ page we will get 404 error

PS
replacing the string with something like
location ~ /\.ht[a-z]+$ {return 404;}
sort of like should solve the problem

**Comment by blondie101010** [comment]
Sorry but your example is an excellent reason to leave it as is.

**Comment by anthonyrossbach** [comment]
I don't see that as a problem at all, the problem will be any code that writes temp or on file version control versions of a file.

So for example if a code replaces a htaccess file and backups the old one incase something breaks, or even if a user does this with a complex htacccess file and it shows paths that are important and should not be public and other info, now the file can be loaded via /.htaccess.old that's a problem also.

The solution would be to use this for your url. example.com/htaccess/some-htaccess-related-topic/

**Comment by scorp13** [comment]
@anthonyrossbach сhanging the URL is certainly a solution, but in my case it cannot be used. I solved my problem (in my case no .htaccess.old-like files are expected), but I agree that it is not suitable for all cases.
Perhaps something like this will do
location ~ /\.ht[^/]+$ {return 404;}
or even more specifically
location ~ /\.(htaccess|htpasswd)$ {return 404;}

My main thought is that if this rule is designed to protect files, it is not entirely correct when it triggered for an URL. The dots in the URL, although rarely found, are not incorrect characters.

**Comment by noogen** [comment]
@scorp13 You still don't understand.   @blondie101010 is saying that there should not be any hidden file exposed to the web PERIOD.   @anthonyrossbach is saying that you should reconsider your design.  Otherwise, you will expose yourself to security issue.  The actual rule should be:

```
    location ~ /\.(?!well-known).* {
        deny all;
        access_log off;
        log_not_found off;
    }
```

Example, beside .htaccess or htpasswd, most php application could have '.env' file.   Examples:
* .env, .env.test, env.production and without blocking access, they could expose database and other credentials
* .git, .svn directory etc...

Ref: https://serverfault.com/questions/755662/nginx-disable-htaccess-and-hidden-files-but-allow-well-known-directory/849537#849537

**Comment by scorp13** [comment]
@noogen you also did not seem to understand what I wrote. Or maybe I vaguely put it.

I do not propose to open access to hidden files.
I suggest changing the line
location ~ /\.ht    {return 404;}, 
which triggers not only for .htaccess or other .ht ... files, but also for any other queries containing ".ht"
for a line that would work ONLY FOR FILES.
I may have provided incorrect variations.

You write that beside .htaccess or htpasswd, most php application could have '.env' file.
So why these files are not blocked by the Vesta panel? In the current version there is only such a code.

location ~ /\.ht    {return 404;}
location ~ /\.svn/  {return 404;}
location ~ /\.git/  {return 404;}
location ~ /\.hg/   {return 404;}
location ~ /\.bzr/  {return 404;}

To summarize:
I do not think this issue is important and would not want to insist on doing, so if contributors consider that the current state is normal, please close the issue.
